### PR TITLE
Fix LDAP user group related issues

### PR DESCRIPTION
**Pull request by stonezdj**
User groups should not have same DN
Should not import an LDAP group if there is a user group with same DN

### CODE REVIEW

File: src/ui/api/projectmember.go [hunk 1]
```diff
@@ -121,11 +135,28 @@ func (pma *ProjectMemberAPI) Post() {
 	projectID := pma.project.ProjectID
 	var request models.MemberReq
 	pma.DecodeJSONReq(&request)
-	pmid, err := AddOrUpdateProjectMember(projectID, request)
+	request.MemberGroup.LdapGroupDN = strings.TrimSpace(request.MemberGroup.LdapGroupDN)
+	//Check valid DN specified
+	if len(request.MemberGroup.LdapGroupDN) > 0 {
+		if _, err := goldap.ParseDN(request.MemberGroup.LdapGroupDN); err != nil {
+			pma.HandleBadRequest("Invalid LDAP DN specified.")
```
Comment (reasonerjt) [inline src/ui/api/projectmember.go]: can we log the error?

File: src/ui/api/projectmember.go [hunk 2]
```diff
@@ -179,18 +210,27 @@ func AddOrUpdateProjectMember(projectID int64, request models.MemberReq) (int, e
...
+		//Check duplicate LDAP DN in usergroup, if usergroup exist, return error
+		userGroupList, err := group.QueryUserGroup(models.UserGroup{LdapGroupDN: request.MemberGroup.LdapGroupDN})
+		if err != nil {
+			return 0, err
+		}
+		if len(userGroupList) > 0 {
+			return 0, ErrDuplicateLDAPGroup
+		}
+		//If groupname provided, use the provided groupname to name this group
 		groupID, err := auth.SearchAndOnBoardGroup(request.MemberGroup.LdapGroupDN, request.MemberGroup.GroupName)
```
Comment (reasonerjt) [inline src/ui/api/projectmember.go]: I think we should fail in `SearchAndOnBoard` or let SearchAndOnBoard in ldap package ensure there's no duplicate entry?

File: src/ui/api/projectmember.go (hunk 1 above)
Comment (stonezdj) [reply src/ui/api/projectmember.go]: pma.HandleBadRequest will log the error

File: src/ui/api/projectmember.go (hunk 2 above)
Comment (stonezdj) [reply src/ui/api/projectmember.go]: OK

**Pull request merged**